lib/module.js

```diff
@@ -218,14 +218,8 @@ Module._findPath = function(request, paths, isMain) {
     if (!filename && rc === 1) {  // Directory.
       if (exts === undefined)
         exts = Object.keys(Module._extensions);
-      filename = tryPackage(basePath, exts, isMain);
-    }
-
-    if (!filename && rc === 1) {  // Directory.
-      // try it with each of the extensions at "index"
-      if (exts === undefined)
-        exts = Object.keys(Module._extensions);
-      filename = tryExtensions(path.resolve(basePath, 'index'), exts, isMain);
+      filename = tryPackage(basePath, exts, isMain) ||
+        tryExtensions(path.resolve(basePath, 'index'), exts, isMain); // try it with each of the extensions at "index"
```

Comment:
Please run `make list` - it should warn that this line is too long with the comment. :)